packages/compat/src/compat-app.ts

```diff
@@ -317,10 +317,6 @@ class CompatAppAdapter implements AppAdapter<TreeNames> {
     return this.configTree.readConfig().rootURL;
   }
 
-  templateCompilerPath(): string {
-    return 'ember-source/vendor/ember/ember-template-compiler';
```

Comment:
It seems like this is actually trying to resolve the path relative to the build itself. That is why it is `ember-source/vendor/ember-template-compiler.js` instead of where it actually is in the npm package `ember-source/dist/ember-template-compiler.js`.

This is where that is done:

https://github.com/emberjs/ember.js/blob/eff00a35624cd33f9f46f0991252182c367bb045/lib/index.js#L298-L300